pytorch_lightning/trainer/connectors/accelerator_connector.py

```diff
@@ -475,6 +474,15 @@ def _set_parallel_devices_and_init_accelerator(self) -> None:
             accelerator_class = ACCELERATORS[self._accelerator_flag]
             self.accelerator = accelerator_class()  # type: ignore[abstract]
 
+        if not self.accelerator.is_available():
```

Comment:
depending on the accelerator, we might not be able to instantiate it without it being available. The init should have the freedom to assume it is available or raise additional errors. IMO the check should have been before the instantiation.